superset/jinja_context.py

```diff
@@ -186,6 +188,28 @@ def url_param(
         return result
 
 
+def safe_proxy(func: Callable[..., Any], *args: Any, **kwargs: Any) -> Any:
```

Comment:
Wondering if we should recurse into lists & dicts here if we're trying to make sure that we have only native/harmless types. Another way to do this would be to try `json.loads(json.dumps(obj))`.

Reply:
I'm not even sure this proxying is strictly necessary, but the Jinja sandbox seems to be handling dangerous function arguments just fine:
```
> SELECT "{{url_param(''.__class__.mro()[1].__subclasses__())}}"`
> Query 306: Template syntax error: access to attribute '__class__' of 'str' object is unsafe.
```

Reply:
I tried the proxy approach in `ipython` to see if someone was to do something like this:
```python
import uuid
def f():
    return None
f.uuid = uuid
```
and the proxying gets rid of that attached attribute.

Reply:
But the main comment was around recursion, if we're trying to make sure that what the function returns is harmless, we'd have to look into list and dicts too, and everything nested under those.

Reply:
Yep, point taken on recursion. The user will have a static set of exposed functions, which could be nested in dicts or lists to little effect. If lamdas could be defined, that would be more problematic, but `{{ (lambda x: x + 1)(2) }}` doesn't seem to work.

Reply:
Another point here is that the functions being called via `safe_proxy` are not user-generated, and we can be fairly certain they are safe due to the existing code review process. I added some logic in my last commit to check top-level context variables for disallowed types. This could potentially benefit from recursion...

Reply:
Recursive data handling added via json serialization, as you suggested.

Reply:
Looks pretty bullet proof to me now!